### PR TITLE
App Builder - Unit and Integration tests

### PR DESCRIPTION
**Pull request by eshurakov**
This pull request introduces improvements to the Cloudflare App Builder project focusing on enhanced test coverage.

**Testing & CI Improvements**
- Added integration tests for the App Builder. 

**API Schema Additions**
- Introduced new API schemas for the `/tree` and `/blob` endpoints, defining types and validation for tree and blob responses.

### CODE REVIEW

File: cloudflare-app-builder/src/git-repository-do.ts [hunk 1]
```diff
@@ -263,4 +263,144 @@ export class GitRepositoryDO extends DurableObject<Env> {
...
     });
   }
+
+  /**
+   * Get directory tree contents at a specific path (RPC method)
+   * @param ref - Branch name (e.g., "main", "HEAD") or commit SHA
+   * @param path - Optional path to a subdirectory (defaults to root)
+   * @returns Array of tree entries with name, type, oid, and mode
+   */
+  async getTree(
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/git-repository-do.ts:278]: This code is duplicated, can we reuse the git implementation from here?

File: cloudflare-app-builder/src/git-repository-do.ts [hunk 2]
```diff
@@ -263,4 +263,144 @@ export class GitRepositoryDO extends DurableObject<Env> {
...
+    });
+  }
+
+  /**
+   * Get blob (file) contents at a specific path (RPC method)
+   * @param ref - Branch name (e.g., "main", "HEAD") or commit SHA
+   * @param path - Path to the file
+   * @returns File content (utf-8 or base64 encoded), encoding type, size, and blob SHA
+   */
+  async getBlob(
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/git-repository-do.ts:306]: Also duplicated and the return types are mismatching?

File: cloudflare-app-builder/src/handlers/files.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      path,
+      error: error instanceof Error ? error.message : String(error),
+    });
+    return new Response(
+      JSON.stringify({
+        error: 'not_found',
+        message: error instanceof Error ? error.message : 'Path not found',
+      }),
+      {
+        status: 404,
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/handlers/files.ts:74]: This will return 404 also for 'Error(`Path not found: ${normalizedPath}`)' and 'Error(`Path is not a directory: ${normalizedPath}`)'. It depends on the quality we want for this API, but an optional improvement would be to handle not found cases explicitly by cascading the error from `getTree`

File: cloudflare-app-builder/src/handlers/files.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      path,
+      error: error instanceof Error ? error.message : String(error),
+    });
+    return new Response(
+      JSON.stringify({
+        error: 'not_found',
+        message: error instanceof Error ? error.message : 'File not found',
+      }),
+      {
+        status: 404,
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/handlers/files.ts:133]: See comment above. Same thing here.

File: cloudflare-app-builder/package.json [hunk 5]
```diff
@@ -18,13 +20,16 @@
...
   },
   "devDependencies": {
-    "@types/jsonwebtoken": "^9.0.9",
     "@cloudflare/containers": "^0.0.30",
     "@cloudflare/sandbox": "0.6.7",
     "@cloudflare/workers-types": "^4.20251014.0",
+    "@types/better-sqlite3": "^7.6.13",
+    "@types/jsonwebtoken": "^9.0.9",
     "@types/node": "^22.10.1",
+    "better-sqlite3": "^12.6.0",
```
Comment (marius-kilocode) [inline cloudflare-app-builder/package.json:29]: We can already use: `12.6.2`, same applies for the other new deps and their patch releases.

File: cloudflare-app-builder/src/git/test-utils.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Close all databases opened during tests
+ */
+export function closeAllDatabases(): void {
+  for (const db of databases) {
+    try {
+      db.close();
+    } catch {
+      // Ignore - might already be closed
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/git/test-utils.ts:68]: Should we not log this?

File: cloudflare-app-builder/src/git/test-utils.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
+/**
+ * Test utilities for GitVersionControl unit tests
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/git/test-utils.ts:2]: The code does't appears really domain specific to git. It rather looks like common code that should be sharable and reusable?

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -250,6 +253,37 @@ jobs:
...
+          cache: 'pnpm'
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Typecheck (app-builder)
+        run: pnpm --filter app-builder typecheck
+
+      - name: Run app-builder tests
+        run: pnpm --filter app-builder test
```
Comment (marius-kilocode) [inline .github/workflows/ci.yml:285]: I am a bit confused here should we not run the integration tests?

**Comment by eshurakov** [comment]
thank you @marius-kilocode, PR is ready for another pass

File: cloudflare-app-builder/src/_integration_tests/run-all-tests.sh [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+  local test_file="$2"
+
+  echo ""
+  echo "========================================"
+  echo "Running: $test_name"
+  echo "========================================"
+  echo ""
+
+  if pnpm exec tsx "$test_file"; then
+    TOTAL_PASSED=$((TOTAL_PASSED + 1))
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/_integration_tests/run-all-tests.sh:42]: Is there a reason why we do all of this manually? Should we not use a proper test runner instead?

File: cloudflare-app-builder/src/_integration_tests/test-init-edge-cases.ts [hunk 10]
```diff
@@ -0,0 +1,371 @@
...
+
+  // Test various special characters - most should work since pattern is [a-z0-9_-]
+  const validTestId = `test-init_special-${Date.now()}-${Math.random().toString(36).slice(2, 8)}`;
+
+  try {
+    const result = await initProjectRaw(validTestId);
+
+    if (result.status === 201 && result.body.success === true) {
+      logSuccess('Init with valid special chars (underscore, hyphen) succeeds');
+      return true;
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/_integration_tests/test-init-edge-cases.ts:281]: I don't get why we are returning true or false here? We need assertions here or not and a proper test framework to detect failures.

File: cloudflare-app-builder/src/handlers/files.ts [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+ * File tree and blob endpoint handlers
+ * GET /apps/{app_id}/tree/{ref} - Get directory contents
+ * GET /apps/{app_id}/blob/{ref}/{path} - Get file content
+ */
+
+import { verifyBearerToken } from '../utils/auth';
+import { logger } from '../utils/logger';
+import type { Env } from '../types';
+
+const notFoundPatterns = [
```
Comment (marius-kilocode) [inline cloudflare-app-builder/src/handlers/files.ts:11]: This is a really hacky model reward hack, once any error string changes the logic changes. We should introduce either proper error codes or error implementations.

File: cloudflare-app-builder/package.json (hunk 5 above)
Comment (marius-kilocode) [reply cloudflare-app-builder/package.json:29]: Still the old version?
